scripts/build.py

```diff
@@ -1351,14 +1353,15 @@ def _build_engine_cmd(self, skip_tests, skip_codesign, disable_ccache, generate_
         return '%s %s/ext/bin/waf --prefix=%s %s %s %s %s %s' % (' '.join(self.get_python()), self.dynamo_home, prefix, skip_tests, skip_codesign, disable_ccache, generate_compile_commands, commands)
 
     def _build_engine_lib(self, args, lib, platform, skip_tests = False, dir = 'engine'):
-        self._log('Building %s for %s' % (lib, platform))
+        self.build_tracker.start_component(lib, platform)
```

Comment:
I think having steps related logs in build_tracker is clearer and less code duplications. If you track it - you log it.